File: metamorph.java.parser/src/java/metamorph.java/DefaultJavaFileProcessor.java
package metamorph.java;

import metamorph.parsing.PrintingListener;
import org.antlr.v4.runtime.*;

import java.io.IOException;
import java.io.PrintWriter;

public final class DefaultJavaFileProcessor implements JavaFileProcessor {

    public void processFile(String filename) {
        ANTLRInputStream input = null;
        try {
            input = new ANTLRFileStream(filename);
            sourceToAst(input, new PrintWriter(System.out));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public String sourceToAst(ANTLRInputStream input, PrintWriter out) {
        JavaLexer lexer = new JavaLexer(input);
        CommonTokenStream tokens = new CommonTokenStream(lexer);
        JavaParser parser = new JavaParser(tokens);
        ParserRuleContext<Token> tree = parser.compilationUnit();
        out.println(tree.toStringTree(parser));

        PrintingListener printingListener = new PrintingListener();

        return printingListener.toStringTree(tree, parser);

//        ParseTreeWalker walker = new ParseTreeWalker(); // create standard walker
//        walker.walk(tree, new PrintingListener(out));
// ExtractInterfaceListener extractor = new ExtractInterfaceListener(parser);
// walker.walk(extractor, tree); // initiate walk of tree with listener
    }
}

File: metamorph.java.parser/src/scala/metamorph/java/DefaultJavaFileProcessor.scala
package metamorph.java

import _root_.java.io.IOException
import org.antlr.v4.runtime.{CommonTokenStream, ANTLRFileStream, ANTLRInputStream}
import metamorph.metamorph.parsing.PrintingListener

final class DefaultJavaFileProcessor extends JavaFileProcessor {

  def processFile(filename: String) {
    try {
      sourceToAst(new ANTLRFileStream(filename))
    }
    catch {
      case e: IOException => {
        e.printStackTrace
      }
    }
  }

  def sourceToAst(input: ANTLRInputStream): String = {
    val parser = new JavaParser(new CommonTokenStream(new JavaLexer(input)))
    val tree = parser.compilationUnit

    PrintingListener.toStringTree(tree, parser)
  }

  def visitSource[T](input: ANTLRInputStream, visitor: JavaBaseVisitor[T]): T = {
    val parser = new JavaParser(new CommonTokenStream(new JavaLexer(input)))

    visitor.visit(parser.compilationUnit())
  }
}